### PR TITLE
For some X, choose a consistent spelling between non-X vs. nonX.

### LINKED ISSUE #440: nonX vs. non-X and similar hyphenation issues
The standard is inconsistent about using "nonzero" and "non-zero"

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #440.

### CODE REVIEW

**Comment by jensmaurer** [comment]
Note: This does not yet address nonempty vs. nonempty.

**Comment by tkoeppe** [comment]
Rebase and rescan?

**Comment by jensmaurer** [comment]
Rebased and added two more fixes.

File: source/std.tex [hunk 1]
```diff
@@ -72,6 +72,7 @@
 %%--------------------------------------------------
 %% add special hyphenation rules
 \hyphenation{tem-plate ex-am-ple in-put-it-er-a-tor name-space name-spaces}
+\hyphenation{non-ze-ro}
```
Comment (tkoeppe) [inline source/std.tex]: a) please add to the existing command above,
b) just `non-zero` maybe?

Comment (tkoeppe) [inline source/std.tex]: c) Add `[std]` to the commit message here.

Comment (jensmaurer) [inline source/std.tex]: a-c done. Note I found "non-ze-ro" in the online Merriam-Webster, but non-zero is good enough for our purposes, I'd guess.

Comment (tkoeppe) [inline source/std.tex]: Indeed. You can't weigh the hints, so I'd much rather only give the good hint, and if there's really a line that needs hyphenation at the second point, I'd provide that as a manual hint on a one-off basis.

**Pull request merged**